### ISSUE TITLE
How to process the training set if it contains anomalous samples ?

### ISSUE THREAD

**Issue by zhangsunny** [decisionintelligence/CATCH#14]
Thank you for open-sourcing your work. I noticed that dividing the dataset according to the **train_lens** specified in the **DETECT_META.csv** file may result in the training portion of some datasets (such as CalIt2 and CICIDS) including anomalous samples, which does not meet the requirement of anomaly-free training. May I ask if you have removed these anomalous points, or were they directly included in the training process?

**Comment by ccloud0525** [comment]
> Thank you for open-sourcing your work. I noticed that dividing the dataset according to the **train_lens** specified in the **DETECT_META.csv** file may result in the training portion of some datasets (such as CalIt2 and CICIDS) including anomalous samples, which does not meet the requirement of anomaly-free training. May I ask if you have removed these anomalous points, or were they directly included in the training process?

Thank you for your question. First of all, there is no rule that training must be free of anomalies. Some papers have proven that including a small number of anomalies during training leads to better results, such as TSB-UAD.

**Comment by zhangsunny** [comment]
Thank you for your response. The detailed and comprehensive evaluation results in your paper are very enlightening to me. I look forward to your insights on the following two points:

1. If I need to use anomaly-free training data and filter out anomalies from the training set during preprocessing, would this approach be fair? I’ve observed that removing training anomalies leads to improved metrics.  

2. For anomaly detection tasks, which often involve severe class imbalance, would using AUC-PR be more appropriate than AUC-ROC ? Papers such as "The Elephant in the Room: Towards a Reliable Time Series Anomaly Detection Benchmark" and "Quo Vadis, Unsupervised Time Series Anomaly Detection?" discuss related issues. What are your views on these metrics?

**Comment by qiu69** [comment]
Thank you for your question！

Regarding the first question, I believe that the dataset should not be changed because we cannot guarantee that there will be no abnormalities in the training process of the dataset in real-world application scenarios. If everyone trains on the same dataset, we can also see the robustness and generalization of our algorithms. Our dataset comes from the [TAB](https://arxiv.org/pdf/2506.18046) benchmark.

For the second question, we reported these metrics simultaneously, and the function of each metric is different. We should look at the performance of the model in the case of multiple metrics. If there are specific requirements in a particular scenario, we can look at the performance of specific metrics.

**Comment by zhangsunny** [comment]
Thank you both for your kind and detailed responses. Looking forward to seeing the new work from you and your lab!

**Comment by qiu69** [comment]
Thank you very much!
